tools/pdf_util.py

```
@@ -0,0 +1,131 @@
+
+
+def get_subtitles(fpth: str):
+    def escape(s: Union[str, bytes]) -> str:
+        ESC_PAT = re.compile(r'[\000-\037&<>()"\042\047\134\177-\377]')
+        if isinstance(s, bytes):
+            us = str(s, "latin-1")
+        else:
+            us = s
+        return ESC_PAT.sub(lambda m: "&#%d;" % ord(m.group(0)), us)
+    #outfp: TextIO,
+    fp = open(fpth, "rb")
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)